packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -665,6 +665,35 @@ export class AtomicSearchBox {
     this.updateSuggestionElements(suggestedQuery);
   }
 
+  private renderPanel(
+    side: 'left' | 'right',
+    elements: SearchBoxSuggestionItem[],
+    setRef: (el: HTMLUListElement | undefined) => void,
+    getRef: () => HTMLUListElement | undefined
+  ) {
+    if (!elements.length) {
+      return null;
+    }
+
+    return (
+      <ul
+        part={`suggestions suggestions-${side}`}
+        aria-label={this.bindings.i18n.t('query-suggestion-list')}
+        ref={setRef}
+        class="flex-grow"
+        onMouseDown={(e) => {
+          if (e.target === getRef()) {
```

Comment:
this `onMouseDown` function is the main hero of this PR, making sure that clicking on the panel doesn't blur the input!

Reply:
Ok ! Hummm I'm wondering if that's going to be "easy" to break down the line.

When we miss clicks analytics recording, it's always causing big problems 😅

What was the problem before this change ? Was it that you could not click on the result ? Like, you would not get redirected to the product itself ? Or just the analytics function that would not properly execute ?

Also, is there the same problem if we use keyboard navigation (ie: tab to the result link on the preview, and press enter) ?

Reply:
https://user-images.githubusercontent.com/30511433/173049524-a6459bc4-c445-4f7c-9977-059c3755f66b.mov

this is a video that shows what happens before this PR. it does log the clicks, but doesn't let the page redirect... 
in general, I think that if an input creates a little popup like that, anywhere you click inside it should leave it open (i.e. clicking on the white space after the queries) unless specifically specified otherwise (like a close button, or, say, updating the query bc we want the results underneath to be immediately visible). 

What this function does, as you might have noticed, is to handle this case in particular, where the click happens on the `ul` where it doesn't have a child. It avoids the input blurring when you click that (and that's why I used mouse down and not mouse over - the blur happens between the mouse down and click events).

Notice the focus outline on this video:
https://user-images.githubusercontent.com/30511433/173050405-941b7be2-619b-4f3a-af14-74f89d08bbc1.mov
for some reason, clicking the `li` elements doesn't remove the focus from the input, but clicking the `ul` does. So to avoid the popup disappearing in both cases, this PR essentially does two things:
- avoids blurring input when clicking the `ul` element
- doesn't clearSuggestions [when a suggestion that has a query is clicked](https://github.com/coveo/ui-kit/pull/2066/files#diff-46e3c6b20a52a198d73e328db1b88c9a1b4b44ba84c5e7556011e630fc8f4638R580) (nothing is prevented in that case)

So, summarising, it don't think it will affect any analytics! 
Thanks for coming to my ted talk! Let me know if you still have questions/concerns!

Reply:
it looks like when you press enter on the result, it doesn't navigate to the page, but only handles the input stuff with the query that's already there. what do you think @btaillon ? any ideas?

https://user-images.githubusercontent.com/30511433/173125978-c74f442e-e854-413e-8769-f798593d7a8f.mov

Reply:
ideally it would reuse the logic from the `LinkWithResultAnalytics` funcitonal component

```
onClick={(e) => stopPropagationAndProcess(e, onSelect)}
onContextMenu={(e) => stopPropagationAndProcess(e, onSelect)}
onMouseDown={(e) => stopPropagationAndProcess(e, onSelect)}
onMouseUp={(e) => stopPropagationAndProcess(e, onSelect)}
onTouchStart={(e) => stopPropagationAndProcess(e, onBeginDelayedSelect)}
onTouchEnd={(e) => stopPropagationAndProcess(e, onCancelPendingSelect)}
```